bindings/python/pymongoarrow/api.py

```diff
@@ -279,17 +292,23 @@ def write(collection, tabular):
     :Returns:
       An instance of :class:`result.ArrowWriteResult`.
     """
-
+    tab_size = len(tabular)
     if isinstance(tabular, Table):
         _validate_schema(tabular.schema.types)
     elif isinstance(tabular, DataFrame):
         _validate_schema(ArrowSchema.from_pandas(tabular).types)
+    elif (
+        isinstance(tabular, dict)
+        and len(tabular.values()) >= 1
+        and isinstance(list(tabular.values())[0], ndarray)
+    ):
+        _validate_schema([i.dtype for i in tabular.values()])
+        tab_size = len(list(tabular.values())[0])
```

Comment:
The list() call here is wasteful since we only need the first value. How about `len(next(iter(tabular.values())))`?